solc/CommandLineInterface.cpp

```diff
@@ -1207,6 +1207,8 @@ void CommandLineInterface::outputCompilationResults()
 	{
 		if (needsHumanTargetedStdout(m_args))
 			cout << endl << "======= " << contract << " =======" << endl;
+		else
+			cout << "Source files are valid, but no output requested." << endl;
```

Comment:
I fear this message will be printed multiple times if you have more than one contract in the source files. It might perhaps be better to set a flag after option parsing or so - perhaps add a helper function like `needsHumanTargetedStdout`.

Reply:
modified code snippet, please review before submission
`
	bool isOutputRequested = true;

	vector<string> contracts = m_compiler->contractNames();
	for (string const& contract: contracts)
	{
		if (needsHumanTargetedStdout(m_args)) {
			cout << endl << "======= " << contract << " =======" << endl;
			isOutputRequested = false;
		}

		if(isOutputRequested) {
			cout << "Source files are valid, but no output requested." << endl;
			isOutputRequested = false;
		}
`

Reply:
Please just force-push an update to your branch, no need to put the code in a comment before submitting.

Could you do it in a way where it happens outside of the loop? Essentially `needsHumanTargetedStdout` does not really depend on the contract in question. Also there are modes where files are generated (and `needsHumanTargetedStdout` returns false), in which case the message should be different - something like "Requested files created".

Reply:
Should I ask my queries here based on your latest comment or Gitter?
By the way, this code was working for me.

Reply:
Since most of the team is currently attending a conference, it might be better to ask questions here.

Reply:
I was able to solve the issue with multiple contracts but having following issue.
Not able to differentiate between 2 commands below:
`./solidity/build/solc/solc -o ./ SimpleContract.sol` and
`/solidity/build/solc/solc -o ./ --userdoc SimpleContract.sol`, 
In my view, `./solidity/build/solc/solc -o ./ SimpleContract.sol` this command also should output
"Source files are valid, but no output requested.".
In my understanding `needsHumanTargetedStdout` is returning `false` in 2 conditions 
1)  in case of g_argOutputDir and
2) for all other options in loop.
I am not able to separate the 2 false conditions `outputCompilationResults`. 

My opinion/suggesttion: can we change `needsHumanTargetedStdout` to return integers instead of bool. It might be relatively easy to fix this issue.

Appreciate your response.

Reply:
If you use `-o`, then `needsHumanTargetedOutput` should return zero, since it does not generate output on stdout that is supposed to be read by humans.

You can change this helper function, but I think a fully new helper function is needed, because the purpose is rather different.